### PR TITLE
Implement lint warning for non-safe NavController.navigate call

### PR DESCRIPTION
**Pull request by fknives**
Since it's easy to forget when to use safeNavController for navigate calls, a lint warning is created which warns you about findNavController.navigate() calls if they are not surrounded with currentDestination check.

Usage is in the readme.md, Thanks for the great suggestion @zsoltboldizsar !

### CODE REVIEW

File: navigation-lint/src/test/java/com/halcyonmobile/android/common/extensions/navigation/lint/UnsafeUnsafeNavigationIssueRegistryTest.kt [hunk 1]
```diff
@@ -0,0 +1,131 @@
+package com.halcyonmobile.android.common.extensions.navigation.lint
+
+import com.android.tools.lint.checks.infrastructure.TestFiles
+import com.android.tools.lint.checks.infrastructure.TestLintTask
+import com.android.tools.lint.detector.api.TextFormat
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+
+class UnsafeUnsafeNavigationIssueRegistryTest {
```
Comment (zsoltboldizsar) [inline navigation-lint/src/test/java/com/halcyonmobile/android/common/extensions/navigation/lint/UnsafeUnsafeNavigationIssueRegistryTest.kt]: Awesome how you used the test files and expected output 👍

**Comment by fknives** [comment]
> We have our first Android lint rule in the history of Halcyon 🎉 Great implementation 👏 .
> 
> As for some minor feedback: we could create a more generic artifact named `lint-checks`, `lints` or something that represents a collection of custom lint rules we have at Halcyon. This could give us the flexibility to warn consumers of possible issues in their codebase without them having to know about any of the upcoming libraries we might add to `android-common-extensions` (in this case we could even suggest them to use `safe.navigation` which they might not be aware of).
> 
> Also whenever we add a new rule consumers will be notified about the new version this way could more easily target them with any coding standard we might introduce in the future.
> 
> What do you think?

I like the idea, however since the safeNavController has a separate artifact I thought it made sense to have one for the lint warning as well (so if someone is interested only in this one, may use only that).
So my approach to this would be the same as for the non-lint module: have separate module if it makes sense and create one which includes every lintcheck into itself via "api" or something similar.
you can see "all" for non-lint modules here: https://github.com/halcyonmobile/android-common-extensions/pull/3

Does this makes sense to you or would you agree to a solution like that?

**Comment by zsoltboldizsar** [comment]
I think I'd rather go for a library holding all the custom lint rules instead of adding two dependencies in my project. Also I can't see myself choosing only a subset of lint-rules but pull in everything and benefit from them right away.

If you're thinking from a Non-Halcyon consumer's point of view, then a standalone lint-rule library wouldn't make sense as we might enforce company specific rules/standards with them, so maybe it would be better to include them directly into `safe.navigation`. What do you think?

**Comment by fknives** [comment]
> I think I'd rather go for a library holding all the custom lint rules instead of adding two dependencies in my project. Also I can't see myself choosing only a subset of lint-rules but pull in everything and benefit from them right away.
> 
> If you're thinking from a Non-Halcyon consumer's point of view, then a standalone lint-rule library wouldn't make sense as we might enforce company specific rules/standards with them, so maybe it would be better to include them directly into `safe.navigation`. What do you think?

I am unaware of the possibility to include the lint rules in the safe.navigation, since you need "lintChecks" not "implement" in gradle.
 
I see your point of not wanting to include two dependencies, but it is possible to have one which includes the navigation lint within (via api dependency) so you would need to include only that one and you get all the lint rules (navigation and others). As I stated i exactly did the same in pull request 3, where adding one dependency you get both the navigation extensions and the view extensions.

Why I think it makes sense to also have a separate lint-check for navigation (not including other custom lints we want to add later) is if we are working with a project with other teams / other structure, we can still use the navigation lint check which is not really enforces any custom rules we have.

**Pull request merged**